[Daniel Lee Min Qiang] iP

dandaandaaaaaan opened the pull request:
### Dusk 🌆 

> **Gather your thoughts and plan your tomorrows with Dusk.**

Task management tool to ensure you never forget something again.

**How to download**
1. Visit the releases [here](https://github.com/dandaandaaaaaan/ip/releases)
2. Download the `.jar` file of the latest release
3. Run the command `java -jar Duke.jar` in the same folder

**Features:**
- [x] Simple task check/unchecking
- [x] Setting deadlines/event timers
- [x] Searching for and saving tasks to external offline locations
- [x] Easy to use GUI 

**Commands**
- `list` Lists all current available tasks
- `mark <taskNumber>` Marks the selected task as completed
- `unmark <taskNumber>` Unmarks the selected task, stays as incomplete.
- `todo <taskTitle>` Add a new todo task
- `deadline <taskTitle> <taskDate> ~<taskTime>~` Add a new deadline task (time is optional)
- `event <eventTitle> <eventDate> ~<eventTime>~` Add a new event (time is optional)
- `delete <taskNumber>` Deletes selected task from the list
- `find <text>` Finds tasks that contain specified text within their title

Here is a preview of the launcher
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

dandaandaaaaaan committed:
Update Duke with Level 1 requirements

dandaandaaaaaan committed:
Added Task class and task listing functionality

dandaandaaaaaan committed:
Merge pull request #1 from dandaandaaaaaan/Level-2

Level 2 functionality

dandaandaaaaaan committed:
Added marking and unmarking functionality to Tasks

dandaandaaaaaan committed:
Merge pull request #2 from dandaandaaaaaan/Level-3

Level 3 Functionality

dandaandaaaaaan committed:
Add DeadlineTask, EventTask, and TodoTask classes

dandaandaaaaaan committed:
Fix modified old getTitle() refs, add todo command

dandaandaaaaaan committed:
Modify list command to fit new requirements

dandaandaaaaaan committed:
Abstract text messages and add deadline and events

dandaandaaaaaan committed:
Abstract task line printing

dandaandaaaaaan committed:
Merge pull request #3 from dandaandaaaaaan/Level-4

Level 4 Functionality

dandaandaaaaaan committed:
Fix slight line spacing issues

dandaandaaaaaan committed:
Add test cases for success flow and invalid input

dandaandaaaaaan committed:
Merge pull request #4 from dandaandaaaaaan/A-TextUiTesting

A-TextUiTesting

dandaandaaaaaan committed:
Add DukeException class and throw errors

dandaandaaaaaan committed:
Update test cases for updated error scenarios

dandaandaaaaaan committed:
Merge pull request #5 from dandaandaaaaaan/Level-5

Level 5 Functionality

dandaandaaaaaan committed:
Add invalid mark error checks

dandaandaaaaaan committed:
Added more marking checks and test cases

dandaandaaaaaan committed:
Merge pull request #6 from dandaandaaaaaan/Level-5

Level 5 - Improvements

dandaandaaaaaan committed:
Cleanup minor duplicating messages

dandaandaaaaaan committed:
Add delete functionality with error handling

dandaandaaaaaan committed:
Add test cases and fix wrong message for delete

dandaandaaaaaan committed:
Merge pull request #7 from dandaandaaaaaan/Level-6

Level 6 Functionality and Tests

dandaandaaaaaan committed:
Use Enums for tasks

dandaandaaaaaan committed:
Merge pull request #8 from dandaandaaaaaan/A-Enums

A-enums

dandaandaaaaaan committed:
Add save to file per manipulation of task list

dandaandaaaaaan committed:
Convert times to LocalDate + LocalTime

dandaandaaaaaan committed:
Merge pull request #9 from dandaandaaaaaan/branch-Level-7

Level 7 - Saving to .txt list

dandaandaaaaaan committed:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/DeadlineTask.java
#	src/main/java/Duke.java
#	src/main/java/EventTask.java

dandaandaaaaaan committed:
Amend saved file path

dandaandaaaaaan committed:
Merge pull request #11 from dandaandaaaaaan/branch-Level-7

Level 7 - Amend saved file path

dandaandaaaaaan committed:
Merge pull request #10 from dandaandaaaaaan/branch-Level-8

Level 8 - Change to use LocalDate and LocalTime

dandaandaaaaaan committed:
Add divider between time and date

dandaandaaaaaan committed:
Merge pull request #12 from dandaandaaaaaan/branch-Level-8

Add divider between time and date

dandaandaaaaaan committed:
Add loading task list from file

dandaandaaaaaan committed:
Merge pull request #13 from dandaandaaaaaan/branch-Level-7

Add loading task list from file

dandaandaaaaaan committed:
Implement more OOP principles throughout

dandaandaaaaaan committed:
Merge pull request #14 from dandaandaaaaaan/A-MoreOOP

A-MoreOOP Implement more OOP principles throughout

dandaandaaaaaan committed:
Simplify invalid command errors

dandaandaaaaaan committed:
Remove DukeInvalidCommandException

dandaandaaaaaan committed:
Package all files accordingly

dandaandaaaaaan committed:
Merge pull request #15 from dandaandaaaaaan/A-Packages

A-Packages, Package all files accordingly

dandaandaaaaaan committed:
Remove unnecessary throws

dandaandaaaaaan committed:
Add JUnit tests for UI, all command and all task

dandaandaaaaaan committed:
Merge pull request #16 from dandaandaaaaaan/A-JUnit

A-JUnit,  Add JUnit tests for UI, all command and all task

dandaandaaaaaan committed:
Fix minor test case typo

dandaandaaaaaan committed:
Add manifest for v0.1 release

dandaandaaaaaan committed:
Add javadoc comments

dandaandaaaaaan committed:
Fix style issues with tests

dandaandaaaaaan committed:
Fix styling issues on source files

dandaandaaaaaan committed:
Add find command functionality

dandaandaaaaaan committed:
Merge pull request #17 from dandaandaaaaaan/branch-A-JavaDoc

A-Javadoc, Add javadoc comments

dandaandaaaaaan committed:
Merge branch 'master' into branch-A-CodingStandard

Resolve all merge conflicts caused from merging of branch-A-JavaDoc with branch-A-CodingStandard

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/ByeCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/CreateCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/Parser.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/task/DeadlineTask.java
#	src/main/java/duke/task/EventTask.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskType.java
#	src/main/java/duke/task/TodoTask.java

dandaandaaaaaan committed:
Merge pull request #18 from dandaandaaaaaan/branch-A-CodingStandard

A-CodingStandard, with conflicts resolved after merging of A-JavaDoc

dandaandaaaaaan committed:
Fix minor nits missed out from conflict resolution

dandaandaaaaaan committed:
Merge pull request #19 from dandaandaaaaaan/branch-A-CodingStandard

Fix minor nits missed out from conflict resolution

dandaandaaaaaan committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/duke/task/Task.java

Added JavaDoc to new method in Task as well

dandaandaaaaaan committed:
Merge pull request #20 from dandaandaaaaaan/branch-Level-9

Level-9, Add find command functionality

dandaandaaaaaan committed:
Fix breaking Ui element due to change of name

dandaandaaaaaan committed:
Merge pull request #21 from dandaandaaaaaan/branch-Level-9

Fix breaking Ui element due to change of name

dandaandaaaaaan committed:
Add javadoc to find and change name

dandaandaaaaaan committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

dandaandaaaaaan committed:
Fix main class name to allow running

dandaandaaaaaan committed:
Merge pull request #22 from dandaandaaaaaan/branch-A-Gradle

A-Gradle, add gradle support

dandaandaaaaaan committed:
Fix test for welcome and align code to checkstyle

dandaandaaaaaan committed:
Merge pull request #23 from dandaandaaaaaan/branch-A-Checkstyle

A-Checkstyle, Fix test for welcome and align code to checkstyle

dandaandaaaaaan committed:
Add JavaFX to gradle build

Cyolune commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,59 @@
+    private static final String ERROR_INVALID_COMMAND = "OOPS!!! You have entered an invalid command :(";
+
+    /**
+     * Parser to process the entered comments.
```
Should the javadocs be in the form of 'Parses...'?

dandaandaaaaaan replied:
thanks for the suggestion! will fix this

Cyolune commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    /**
+     * Execution of the delete command to delete the task.
```
Should this be in the form "Executes ..."?

dandaandaaaaaan replied:
thanks for the suggestion! will fix this

Cyolune commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
```
Would splitting the imports between the io and util blocks make it more readable?

dandaandaaaaaan replied:
I followed the checkstyle to split between static imports, java imports, org imports and then internal package imports

Cyolune commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,82 @@
+ * Class for printing UI elements.
+ */
+public class Ui {
+    private static final String MESSAGE_INTRO = "Hello! I'm Dusk\n     What can I do for you?";
```
Is Dusk a typo? 🤣

dandaandaaaaaan replied:
hahah nope I wanted to change the name to be more creative but many of the classes were still in the old `Duke` name

Cyolune reviewed (commented):
LGTM

honganhcs commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+     * @return 1 if operation if successful
+     */
+    public static int saveToFile(List<Task> taskList) {
+        FileWriter out = null;
```
Perhaps, you can rename `out` to a noun?
Also, would it better to declare and initialize it at the [same place](https://se-education.org/guides/conventions/java/intermediate.html#variables)?

dandaandaaaaaan replied:
thanks for catching this! will declare it in place

honganhcs commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+        //Check if folder/file is present, otherwise create file
+        File file = new File(FILE_PATH);
+        try {
+            String[] split = FILE_PATH.split("/");
```
Perhaps, you can rename `split` to a more descriptive noun? There are a few other places where the variable name can be more descriptive, following the [code quality](https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html) guide 😃

dandaandaaaaaan replied:
thanks for the catch! will rename to something more descriptive

dandaandaaaaaan committed:
Fix minor javadoc inconsistencies

honganhcs commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,82 @@
+     * @param message Template text for the printed task
+     * @return Combined string mainly used for printing
+     */
+    public String taskLine(Task task, String message) {
```
Maybe, since these are methods, you can consider renaming them to start with a verb?

dandaandaaaaaan replied:
thanks for the suggestion! will rename this

honganhcs commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,78 @@
+import duke.task.Task;
+
+public class MarkCommand extends Command {
+    private static final String MESSAGE_MARK = "Nice! I've marked this task as done:";
```
I like that you name related items, e.g. messages similarly 😄

honganhcs reviewed (commented):
Hi, your PR looks good 😄 just a few places where you may want to consider renaming the variables or methods to be more descriptive, following the code quality guide!

dandaandaaaaaan committed:
Resolve styling nits from reviews

dandaandaaaaaan committed:
Implement GUI in place of CLI

dandaandaaaaaan committed:
Cleanup from checkstyle

dandaandaaaaaan committed:
Remove unused methods and fix welcome message

dandaandaaaaaan committed:
Merge branch 'master' into branch-Level-10

# Conflicts:
#	src/main/java/duke/command/MarkCommand.java

dandaandaaaaaan committed:
Cleanup minor nits

dandaandaaaaaan committed:
Merge pull request #24 from dandaandaaaaaan/branch-Level-10

Level-10 Branch, JavaFX GUI

dandaandaaaaaan committed:
Add the use of varargs

dandaandaaaaaan committed:
Merge pull request #25 from dandaandaaaaaan/A-Varargs

A-Varargs, Add the use of varargs

dandaandaaaaaan committed:
Fix wrap text display issues

dandaandaaaaaan committed:
Merge pull request #26 from dandaandaaaaaan/branch-Level-10

Level-10, Fix wrap text display issues

dandaandaaaaaan committed:
Change compiled program name

dandaandaaaaaan committed:
Improve code quality for some methods

Some methods have too many levels, for eg. in CreateCommand.java. Some methods like the one in MarkCommand.java have some duplicated lines

This can be a bit hard to read

Split actions out to separate private methods to make it easier to read, and organise the private variables and make the happy path more obvious

Done this way as per code quality guidelines in order to make the code more readable

dandaandaaaaaan committed:
Add assertions around the code

Current cases mostly handle for user input through the use of exceptions

Assertions are needed in order to spot for any potential programmer errors, especially potential regressions if more features are worked on

Added assertions to various places where it would be important that a property needs to be checked

This is done for easy catching of potential bugs in the code

dandaandaaaaaan committed:
Merge pull request #27 from dandaandaaaaaan/A-Assertions

A-Assertions, Add assertions around the code

dandaandaaaaaan committed:
Merge pull request #28 from dandaandaaaaaan/A-CodeQuality

Improve code quality for some methods

dandaandaaaaaan committed:
Add CI yml file

Currently, there is no configured GitHub Actions for automatic deployment

It would be extremely useful if there were automated building such that further updates can be quickly released

Added gradle file to configure the CI

This is to allow for the built in GitHub Actions to do the necessary automated compiling through Gradle

dandaandaaaaaan committed:
Fix JUnit tests and align checkstyle

JUnit tests were failing after the slight changes to the code when moving from CLI interface to GUI

JUnit tests are important to ensure core functionality of the app can be tested for quickly.

This commit fixes all the tests to align to the changes that were made previously, in order for the automatic github CI to be able to successfully build the project

This is done mainly by checking the returned strings compared to previously done via the stdout printed contents

dandaandaaaaaan committed:
Merge pull request #29 from dandaandaaaaaan/A-CI

A-CI, Configure Github Actions CI

dandaandaaaaaan committed:
Remove and change more old CLI tests

dandaandaaaaaan committed:
Merge pull request #31 from dandaandaaaaaan/A-CI

Remove and change more old CLI tests

dandaandaaaaaan committed:
Add update task functionality

Currently, there is no way to edit individual attributes of a task once it has been added to the list.

This needs to change as it is very possible that a user needs to edit certain details about their task

Added an update command with various params in order to update individual attributes

Done this way to fit with the command theme of the other messages

dandaandaaaaaan committed:
Fix style errors

dandaandaaaaaan committed:
Merge pull request #32 from dandaandaaaaaan/C-Update

C-Update, Add update task functionality

dandaandaaaaaan committed:
Fix code quality issues highlighted by bot

dandaandaaaaaan committed:
Change colour scheme and fix resizing

dandaandaaaaaan committed:
Merge pull request #34 from dandaandaaaaaan/branch-A-BetterGui

A-BetterGUI - Change colour scheme and fix resizing

dandaandaaaaaan committed:
Upload Ui screenshot

dandaandaaaaaan committed:
Merge pull request #35 from dandaandaaaaaan/branch-upload-ui-screenshot

Upload Ui screenshot

dandaandaaaaaan committed:
Add user guide

dandaandaaaaaan committed:
Merge pull request #36 from dandaandaaaaaan/branch-A-UserGuide

A-UserGuide, Add user guide

dandaandaaaaaan committed:
Fix manifests and javafx versions
